floyd/cli/experiment.py

```diff
@@ -164,26 +165,27 @@ def delete(ids, yes):
     """
     Delete project runs
     """
-    
+    failures = False
     for id in ids:
         experiment = ExperimentClient().get(id)
-        task_instance_id = get_module_task_instance_id(experiment.task_instances)
-        task_instance = TaskInstanceClient().get(task_instance_id) if task_instance_id else None
-
+        if not experiment:
+            failures = True
+            continue
 
-        if experiment.state in ["queued", "running"]:
```

Comment:
Yeah good catch. If we just attempt to delete the experiment first, that would also solve the problem, and allow us to decouple the cli from knowing why an experiment might fail to delete. I also added a test for this situation. What do you think of this approach?

It seems impossible to totally remove the risk of having either the experiment and the module be deleted and have the other fail, though, since they are separate calls to the API and either request could fail for a number of different reasons. (Maybe this is something that the API itself should take care of if we never have a reason to delete only one of them).

One problem we have if one fails and the other doesn't is that the cli currently doesn't offer an interface to delete a module directly.